e2e/launcher/pocketserver.go

```diff
@@ -0,0 +1,124 @@
+package launcher
+
+import (
+	"errors"
+	"fmt"
+	"os/exec"
+	"strings"
+	"sync"
+)
+
+type PocketServer interface {
+	Start(...string) error
+	Kill() error
+	RegisterPatternActor(patternActor PatternActor, stream Stream) error
```

Comment:
DISCUSS: I'm familiar with the "Actor Model" from prior work but this is the first I've heard of "Pattern Actor". 

Can you link to any other work that references this naming convention?

Reply:
Reference to the pattern-action paradigm as seen in awk and similar tools. Open to renaming.

Reply:
Let's rename to `patternAction`